src/js/mixins/evented.js

```diff
@@ -0,0 +1,385 @@
+   *         If the first argument was another evented object, this will be
+   *         the listener function; otherwise, _all_ listeners bound to the
+   *         event type(s) will be removed.
+   *
+   * @return {Object}
+   *         Returns the object itself.
+   */
+  off(targetOrType, typeOrListener, listener) {
+
+    // Targeting this evented object.
+    if (!targetOrType || isValidEventType(targetOrType)) {
+      Events.off(this.eventBusEl_, targetOrType, typeOrListener);
+
+    // Targeting another evented object.
+    } else {
+      const target = targetOrType;
+      const type = typeOrListener;
+
+      // Fail fast and in a meaningful way!
+      validateTarget(target);
+      validateEventType(type);
+      validateListener(listener);
+
+      // Ensure there's at least a guid, even if the function hasn't been used
+      listener = Fn.bind(this, listener);
+
+      // Remove the dispose listener on this evented object, which was given
+      // the same guid as the event listener in on().
+      this.off('dispose', listener);
+
+      if (target.nodeName) {
+        Events.off(target, type, listener);
+        Events.off(target, 'dispose', listener);
+      } else if (isEvented(target)) {
+        target.off(type, listener);
+        target.off('dispose', listener);
+      }
+    }
+  },
+
+  /**
+   * Fire an event on this evented object, causing its listeners to be called.
+   *
+   * @param  {string|Object} event
+   *         An event type or an object with a type property.
+   *
+   * @param  {Object} [hash]
+   *         An additional object to pass along to listeners.
+   *
+   * @return {Object}
+   *         Returns this object.
+   */
+  trigger(event, hash) {
+    Events.trigger(this.eventBusEl_, event, hash);
+  }
+};
+
+/**
+ * Makes an object "evented" - granting it methods from the `Events` utility.
+ *
+ * By default, this adds the `off`, `on`, `one`, and `trigger` methods, but
+ * exclusions can optionally be made.
+ *
+ * @param  {Object} target
+ *         The object to which to add event methods.
+ *
+ * @param  {Object} [options={}]
+ *         Options for customizing the mixin behavior.
+ *
+ * @param  {String} [options.eventBusKey]
+ *         By default, adds a `eventBusEl_` DOM element to the target object,
+ *         which is used as an event bus. If the target object already has a
+ *         DOM element that should be used, pass its key here.
+ *
+ * @return {Object}
+ *         The target object.
+ */
+function evented(target, options = {}) {
+  const {eventBusKey} = options;
+
+  // Set or create the eventBusEl_.
+  if (eventBusKey) {
+    if (!target[eventBusKey].nodeName) {
+      throw new Error(`The eventBusKey "${eventBusKey}" does not refer to an element.`);
+    }
+    target.eventBusEl_ = target[eventBusKey];
+  } else {
+    target.eventBusEl_ = Dom.createEl('span', {className: 'vjs-event-bus'});
+  }
+
+  ['off', 'on', 'one', 'trigger'].forEach(name => {
```

Comment:
What about `Objecy.keys(mixin)` that way we're not repeating ourselves? And if we ever add another thing to mixin, it'll automatically get picked up.